databricks/sdk/runtime/__init__.py

```diff
@@ -16,6 +16,18 @@
 except ImportError:
     init_runtime_native_auth = None
 
+if init_runtime_native_auth is None:
+    try:
+        from dbruntime.databricks_repl_context import get_context
+        ctx = get_context()
+        if ctx is not None:
```

Comment:
This is a static check on ctx done on import, but `get_context` will not return a Context until a command has been run. This probably works because databricks.sdk.rutnime is probably not imported until a user interacts with the sdk, but this is pretty brittle and will break if we internally import this file on startup in the future.

Can we make this check either dynamic or at if we want to check once and then give up, could we be more explcit about when the check is done.

Reply:
addressed in https://github.com/databricks/databricks-sdk-py/pull/152/commits/f37186556cd70411c3188f43a42fc797553af811 - we need a hostname during the SDK initialization stage. technically, if we can start every runtime with `DATABRICKS_HOST` environment variable with the f'https://{ctx.browserHostName}', we can simplify the `init_runtime_native_auth` return type. 

please remember - `init_runtime_native_auth` returns the workspace hostname and function that returns a dictionary of headers. this is to support future improvements to REPL authentication, where we could leverage token refreshes